docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,261 @@
+
+### Option A - Continue to use Mustache
+
+- Description
+  - Keep the email templates as is in Mustache
+- Pros
+  - The effort to refactor all the templates would be eliminated for now
+- Cons
+  - The logicless nature of Mustache would continue to bother us in the future
+  - If we continue to use Mustache for our templates then our front-end frameworks in FxA would continue to be different. We currently have Backbone in the content-server, Mustache in the auth-server, and React everywhere else.
+
+### Option B - Use React server-side to generate static HTML templates
+
+- Description
+  - Template the auth server emails using [React](https://reactjs.org/) server-side and convert them over to [static HTML templates](https://reactjs.org/docs/react-dom-server.html#rendertostaticmarkup). Other pros and cons of React can be found [in this ADR](https://github.com/mozilla/fxa/blob/main/docs/adr/0010-transition-fxa-from-backbone-to-react.md#option-c---move-to-isomorphic-react), though the type of React we're describing here would be stateless.
+- Pros
+  - The logicless nature of the templates will be eliminated once they're refactored in React
+  - We already use React in `fxa-settings`, `fxa-admin-panel`, `fxa-payments-server`, and `fxa-react`. There’s potential to cross-share components with the new email templates.
+  - React offers us many integration options. E.g there is one with [MJML](https://github.com/wix-incubator/mjml-react)
+  - React has already been audited by our security team.
+- Cons
+  - It could be a overkill to use React, since the library comes with far too many features which may not be needed for templating
+  - The templates in the `fxa-auth-server` do not contain heavy business logic so building them with a frontend-heavy framework like React might complicate things
+  - Since we will be converting React components into static HTML pages, another layer of complexity as well as new dependencies would be added to the project
+
+### Option C - Use EJS
+
+- Description
+  - Template the auth server mail components using [EJS](https://ejs.co/)
+- Pros
+  - The syntax and logic of EJS is similar to JavaScript, leading to a very small learning curve.
+  - EJS has a large community of users which will ensure faster help and support.
+  - It is a simple templating engine and far easier to use as its plain Javascript and since we want to keep our mail templates straightforward, using EJS could be a good choice.
+- Cons
+  - Adds a new dependency to the stack
+  - Lacks support for `block` functionality, which will make resuing components a bit difficult.
```

Comment:
Using MJML with EJS would address this, as MJML can do components.

Reply:
Nice, yeah, that's worth mentioning like,

> Lacks support for `block` functionality, which will make reusing components a bit difficult, unless we use EJS in tandem with MJML